src/main/java/org/springframework/data/mapping/PersistentProperty.java

```diff
@@ -70,9 +70,21 @@
 	 * {@link Map}'s value type transparently.
 	 *
 	 * @return
+	 * @deprecated since 2.6 for removal in 3.0. Use {@link #getPersistentEntityTypeInformation()} instead.
 	 */
+	@Deprecated
 	Iterable<? extends TypeInformation<?>> getPersistentEntityTypes();
 
+	/**
+	 * Returns the {@link TypeInformation} if the property references a {@link PersistentEntity}. Will return
+	 * {@literal null} in case it refers to a simple type. Will return {@link Collection}'s component type or the
+	 * {@link Map}'s value type transparently.
+	 *
+	 * @return
+	 * @since 2.6
+	 */
+	Iterable<? extends TypeInformation<?>> getPersistentEntityTypeInformation();
+
```

Comment:
The documentation is a bit confusing as it states the method to return `null` for a method thats return type is an `Iterable`. Wouldn't _empty_ be a better fit?

Reply:
Good catch. Indeed, the Javadoc still stems from the times that the original method which the comment was copied from returned a single type, not a collection.